internal/provider/organization_data_source.go

```diff
@@ -90,12 +91,11 @@ func (d *OrganizationDataSource) Read(ctx context.Context, req datasource.ReadRe
 
 	// we fetch the organization id from the service account
 	orgMrn := ""
-	if data.OrgMrn.ValueString() != "" {
+	if data.OrgMrn.ValueString() != "" && data.OrgID.ValueString() == "" {
 		orgMrn = data.OrgMrn.ValueString()
-	} else if data.OrgID.ValueString() != "" {
+	} else if data.OrgID.ValueString() != "" && data.OrgMrn.ValueString() == "" {
 		orgMrn = orgPrefix + data.OrgID.ValueString()
-	}
-	if orgMrn == "" {
+	} else {
```

Comment:
I like to better understand why those conditions have been changed? I am not understanding the problem then the new implementation is fixing.

Reply:
The logic remains the same, I just moved it into an `else` block instead of a separate `if`.